### ISSUE TITLE
PRINT on JSON array topic messages throws com.fasterxml.jackson.databind.node.ArrayNode

### ISSUE THREAD

**Issue by rmoff** [confluentinc/ksql#1933]
```
ksql> print 'zipkin' from beginning;
com.fasterxml.jackson.databind.node.ArrayNode cannot be cast to com.fasterxml.jackson.databind.node.ObjectNode
```

Sample message: 

```
$ docker run --network talk-kafka-zipkin_default --tty --interactive --rm \
          confluentinc/cp-kafkacat \
          kafkacat -b kafka:9092 -C -t zipkin -c 1
[{"traceId":"84f3c1ed60c32913","id":"84f3c1ed60c32913","kind":"PRODUCER","name":"on_send","timestamp":1537797103410968,"duration":44,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"6044c5789a1c65c4","id":"6044c5789a1c65c4","kind":"PRODUCER","name":"on_send","timestamp":1537797103949052,"duration":15,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"13bc1791b1fc86e0","id":"13bc1791b1fc86e0","kind":"PRODUCER","name":"on_send","timestamp":1537797103950100,"duration":13,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"9d4c87e13110ed91","id":"9d4c87e13110ed91","kind":"PRODUCER","name":"on_send","timestamp":1537797103951038,"duration":11,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"c2d9904d52debf47","id":"c2d9904d52debf47","kind":"PRODUCER","name":"on_send","timestamp":1537797104052037,"duration":8,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"df0b9dc98c1c87a4","id":"df0b9dc98c1c87a4","kind":"PRODUCER","name":"on_send","timestamp":1537797104054053,"duration":9,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"ea471a63f72202c8","id":"ea471a63f72202c8","kind":"PRODUCER","name":"on_send","timestamp":1537797104157054,"duration":14,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}}]
```

=> [Pretty-print of the JSON message](https://jsonformatter.org/json-pretty-print/f5f3ec)

1. KSQL should ideally be able to display this
2. At very minimum it should throw a better error message / attempt to render the data as raw string / something nicer.

**Comment by apurvam** [comment]
Looks like it can't print arrays of structs. I assume that a `CREATE STREAM` and then `SELECT *` on that stream works fine?

**Comment by lssilva** [comment]
I have the same issue but with Int

com.fasterxml.jackson.databind.node.IntNode cannot be cast to com.fasterxml.jackson.databind.node.ObjectNode

Creating a stream and selecting works fine for me!

**Comment by rmoff** [comment]
@lssilva are you able to share a sample of your data that causes this error in `PRINT`?

**Comment by rmoff** [comment]
Thanks for the update @lssilva. I'm going to leave this ticket open as the original problem still exists. 

Producer sample record: 

```
$ echo '[{"traceId":"84f3c1ed60c32913","id":"84f3c1ed60c32913","kind":"PRODUCER","name":"on_send","timestamp":1537797103410968,"duration":44,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"6044c5789a1c65c4","id":"6044c5789a1c65c4","kind":"PRODUCER","name":"on_send","timestamp":1537797103949052,"duration":15,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"13bc1791b1fc86e0","id":"13bc1791b1fc86e0","kind":"PRODUCER","name":"on_send","timestamp":1537797103950100,"duration":13,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"9d4c87e13110ed91","id":"9d4c87e13110ed91","kind":"PRODUCER","name":"on_send","timestamp":1537797103951038,"duration":11,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"c2d9904d52debf47","id":"c2d9904d52debf47","kind":"PRODUCER","name":"on_send","timestamp":1537797104052037,"duration":8,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"df0b9dc98c1c87a4","id":"df0b9dc98c1c87a4","kind":"PRODUCER","name":"on_send","timestamp":1537797104054053,"duration":9,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"ea471a63f72202c8","id":"ea471a63f72202c8","kind":"PRODUCER","name":"on_send","timestamp":1537797104157054,"duration":14,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}}] ' | kafkacat -b localhost:9092 -P -t zipkin
```

KSQL: 

```
ksql> VERSION
Version: 5.1.0
ksql> print 'zipkin' from beginning;
com.fasterxml.jackson.databind.node.ArrayNode cannot be cast to com.fasterxml.jackson.databind.node.ObjectNode
ksql>
```

KSQL Server log: 

```
[2019-02-06 10:19:16,174] INFO Building AST for print 'zipkin' from beginning;. (io.confluent.ksql.KsqlEngine:323)
[2019-02-06 10:19:16,191] INFO Printing topic 'zipkin' (io.confluent.ksql.rest.server.resources.streaming.StreamedQueryResource:132)
[2019-02-06 10:19:16,197] ERROR Exception encountered while writing to output stream (io.confluent.ksql.rest.server.resources.streaming.TopicStreamWriter:112)
java.lang.ClassCastException: com.fasterxml.jackson.databind.node.ArrayNode cannot be cast to com.fasterxml.jackson.databind.node.ObjectNode
        at io.confluent.ksql.rest.server.resources.streaming.TopicStream$Format$3$1.print(TopicStream.java:175)
        at io.confluent.ksql.rest.server.resources.streaming.TopicStream$RecordFormatter.lambda$format$1(TopicStream.java:75)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
        at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
        at java.util.Iterator.forEachRemaining(Iterator.java:116)
        at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at io.confluent.ksql.rest.server.resources.streaming.TopicStream$RecordFormatter.format(TopicStream.java:82)
        at io.confluent.ksql.rest.server.resources.streaming.TopicStreamWriter.write(TopicStreamWriter.java:95)
```

**Comment by rmoff** [comment]
Still a problem in 5.2.1

**Comment by vcrfxia** [comment]
Closing as this is no longer a problem on the latest `master`. The new output with @rmoff 's example is:
```
ksql> print 'zipkin' from beginning;
Format:STRING
10/31/19 6:46:46 PM PDT , NULL , [{"traceId":"84f3c1ed60c32913","id":"84f3c1ed60c32913","kind":"PRODUCER","name":"on_send","timestamp":1537797103410968,"duration":44,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"6044c5789a1c65c4","id":"6044c5789a1c65c4","kind":"PRODUCER","name":"on_send","timestamp":1537797103949052,"duration":15,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"13bc1791b1fc86e0","id":"13bc1791b1fc86e0","kind":"PRODUCER","name":"on_send","timestamp":1537797103950100,"duration":13,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"9d4c87e13110ed91","id":"9d4c87e13110ed91","kind":"PRODUCER","name":"on_send","timestamp":1537797103951038,"duration":11,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"c2d9904d52debf47","id":"c2d9904d52debf47","kind":"PRODUCER","name":"on_send","timestamp":1537797104052037,"duration":8,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"df0b9dc98c1c87a4","id":"df0b9dc98c1c87a4","kind":"PRODUCER","name":"on_send","timestamp":1537797104054053,"duration":9,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}},{"traceId":"ea471a63f72202c8","id":"ea471a63f72202c8","kind":"PRODUCER","name":"on_send","timestamp":1537797104157054,"duration":14,"localEndpoint":{"serviceName":"twitter-connector","ipv4":"172.18.0.9"},"remoteEndpoint":{"serviceName":"kafka"},"tags":{"kafka.topic":"twitter_json_01"}}]
```
